apps/search/api/filters/post_resolve_filters.py

```diff
@@ -0,0 +1,73 @@
+from typing import Dict, Union
+
+from rest_framework import filters
+
+from apps.dates.structures import HistoricDateTime
+from apps.search.api.search import SEARCHABLE_DOCUMENTS
+from apps.search.models import SearchableDatedModel
+
+SORT_BY_PARAM = 'ordering'
+
+
+class ApplyMetaFilterBackend(filters.BaseFilterBackend):
+    """
+    Filter that applies meta field from elasticsearch response that contains fields like highlight and score.
+    """
+
+    def __init__(self):
+        self.view = None
+
+    def filter_queryset(self, request, queryset, view):
+        self.view = view
+        return list(map(self.apply_meta, queryset))
+
+    def apply_meta(self, model):
+        document = next(
+            (
+                document
+                for document in SEARCHABLE_DOCUMENTS.values()
+                if isinstance(model, document.django.model)
+            ),
+            None,
+        )
+        index = document.index_name()
+        key = f"{index}_{model.pk}"
+        hit = self.view.search.results_by_id[key]
+        model.meta = hit.meta
+        return model
+
+
+class SortByFilterBackend(filters.BaseFilterBackend):
+    """
+    Filter that sorts queryset by SORT_BY_PARAM
+    """
+
+    def filter_queryset(self, request, queryset, view):
+        sort_by_date = request.query_params.get(SORT_BY_PARAM) == 'date'
+
+        key = date_sorter if sort_by_date else score_sorter
+
+        return sorted(queryset, key=key, reverse=not sort_by_date)
```

Comment:
@alashow, the sorting has to be reverse if it's based on relevance score?

Reply:
Yes. If `reverse=True` is ascending order:
1. date sort sorts from beginning time to end time = `asc`
2. relevance score sorts from highest score to lowest score = `descending`